src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs

```diff
@@ -310,24 +310,34 @@ public partial class UpdateDurableFunction
         }
     }
 
-    private IReadOnlyList<InstanceMetadata> GetInstanceMetadataList(IReadOnlyList<InstanceMetadata> instanceMetadataList)
+    private async Task HandleException(
```

Comment:
This method is already modifying the `UpdateCheckpoint`. Make that more obvious by returning it and you can increment the count there too:

```csharp
private async Task<UpdateCheckpoint> ProcessStudyFailureAsync(
    IDurableOrchestrationContext context,
    UpdateCheckpoint input,
    string studyInstanceUid,
    IReadOnlyList<InstanceMetadata> instances,
    IReadOnlyList<string> instanceErrors)
{
}
```